workos/types/user_management/session.py

```diff
@@ -0,0 +1,44 @@
+from typing import List, Optional, TypedDict, Union
+from enum import Enum
+
+from workos.types.user_management.impersonator import Impersonator
+from workos.types.user_management.user import User
+from workos.types.workos_model import WorkOSModel
+
+
+class AuthenticateWithSessionCookieFailureReason(Enum):
+    INVALID_JWT = "invalid_jwt"
+    INVALID_SESSION_COOKIE = "invalid_session_cookie"
+    NO_SESSION_COOKIE_PROVIDED = "no_session_cookie_provided"
+
+
+class AuthenticateWithSessionCookieSuccessResponse(WorkOSModel):
+    authenticated: bool = True
+    session_id: str
+    organization_id: Optional[str] = None
+    role: Optional[str] = None
+    permissions: Optional[List[str]] = None
```

Comment:
Odd Python-ism, but this can be `Sequence`  since it doesn't necessarily need to be mutable:
```suggestion
    permissions: Optional[Sequence[str]] = None
```